src/client/options/mod.rs

```diff
@@ -675,40 +675,139 @@ impl Serialize for ClientOptions {
+    ///
+    /// The default value is 100.
     pub max_pool_size: Option<u32>,
+
+    /// The minimum number of connections that should be available in a server's connection pool at
+    /// a given time. If fewer than `min_pool_size` connections are in the pool, connections will
+    /// be added to the pool in the background until `min_pool_size` is reached.
+    ///
+    /// The default value is 0.
     pub min_pool_size: Option<u32>,
+
+    /// The amount of time that a connection can remain idle in a connection pool before being
+    /// closed. A value of zero indicates that connections should not be closed due to being idle.
+    ///
+    /// By default, connections will not be closed due to being idle.
     pub max_idle_time: Option<Duration>,
-    pub wait_queue_timeout: Option<Duration>,
+
+    /// The compressors that the Client is willing to use in the order they are specified
+    /// in the configuration.  The Client sends this list of compressors to the server.
+    /// The server responds with the intersection of its supported list of compressors.
+    /// The order of compressors indicates preference of compressors.
     pub compressors: Option<Vec<Compressor>>,
+
+    /// The connect timeout passed to each underlying TcpStream when attemtping to connect to the
+    /// server.
+    ///
+    /// The default value is 10 seconds.
     pub connect_timeout: Option<Duration>,
+
+    /// Whether or not the client should retry a read operation if the operation fails.
+    ///
+    /// The default value is true.
     pub retry_reads: Option<bool>,
+
+    /// Whether or not the client should retry a write operation if the operation fails.
+    ///
+    /// The default value is true.
     pub retry_writes: Option<bool>,
-    pub socket_timeout: Option<Duration>,
-    pub zlib_compression: Option<i32>,
+
+    /// Specifies whether the Client should directly connect to a single host rather than
+    /// autodiscover all servers in the cluster.
+    ///
+    /// The default value is false.
     pub direct_connection: Option<bool>,
+
+    /// The credential to use for authenticating connections made by this client.
     pub credential: Option<Credential>,
+
+    /// Default database for this client.
+    ///
+    /// By default, no default database is specified.
     pub default_database: Option<String>,
+
+    /// Whether or not the client is connecting to a MongoDB cluster through a load balancer.
+    pub load_balanced: Option<bool>,
+
+    pub(crate) srv: bool,
+    pub(crate) wait_queue_timeout: Option<Duration>,
+    pub(crate) socket_timeout: Option<Duration>,
+    pub(crate) zlib_compression: Option<i32>,
     max_staleness: Option<Duration>,
     tls_insecure: Option<bool>,
     auth_mechanism: Option<AuthMechanism>,
     auth_source: Option<String>,
     auth_mechanism_properties: Option<Document>,
     read_preference: Option<ReadPreference>,
     read_preference_tags: Option<Vec<TagSet>>,
```

Comment:
I think we should make some of these public (specifically the ones that are outlined in the [uri-options](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst) spec). Basically anything that is a key in a connection string can be a public option here.

Reply:
Done.